File: backend/src/models/device_outage.py
from pydantic import BaseModel

class DeviceOutageEvent(BaseModel):
    action: str
    handler: str
    at: str
    from_status: str
    to_status: str
    note: str

class DeviceOutage(BaseModel):
    id: int | float
    device_id: int | float
    reason: str
    expected_recovery_at: str
    status: str
    applicant: str
    applied_at: str
    confirmer: str
    confirmed_at: str
    device_status_before: str
    device_status_after: str
    check_result: str
    check_note: str
    checker: str
    checked_at: str
    recoverer: str
    recovered_at: str
    events: list[DeviceOutageEvent]

File: frontend/src/types/DeviceOutage.ts
export interface DeviceOutageEvent {
  action: string;
  handler: string;
  at: string;
  from_status: string;
  to_status: string;
  note: string;
}

export interface DeviceOutage {
  id: number;
  device_id: number;
  reason: string;
  expected_recovery_at: string;
  status: string;
  applicant: string;
  applied_at: string;
  confirmer: string;
  confirmed_at: string;
  device_status_before: string;
  device_status_after: string;
  check_result: string;
  check_note: string;
  checker: string;
  checked_at: string;
  recoverer: string;
  recovered_at: string;
  events: DeviceOutageEvent[];
}

export interface DeviceOutageApplyPayload {
  device_id: number;
  reason: string;
  expected_recovery_at: string;
}

export interface DeviceOutageCheckPayload {
  result: string;
  note: string;
}
